docs/src/pages/components/data-grid/style/StripedGrid.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import { DataGrid, GridRowProps, GridRow, gridClasses } from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+import clsx from 'clsx';
+
+const StripedDataGrid = styled(DataGrid)({
+  [`& .${gridClasses.row}.odd`]: {
+    backgroundColor: '#EEEEEE',
+    '&:hover, &.Mui-hovered': {
```

Comment:
It needs to handle the case `.Mui-selected`. Overwise odd lines do not provide feedback when selected

Reply:
I improved the striped behavior to use the theme and handle `.Mui-selected`